IMPRO-1809: Independent options for lower and upper single value bins

bayliffe opened the pull request:
Added independent options for including lower and upper single value bins for reliability calibration tables. This enables, for example, the inclusion of only the single value bin corresponding with 0 for use calibrating precipitation.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/calibration/reliability_calibration.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5997286&bid=19663972)

Kat-90 commented on improver/cli/construct_reliability_tables.py:
```diff
@@ -64,10 +65,12 @@ def process(
+            precision tolerance, defined as 1.0E-6. The bin is thus 0 to 1.0E-6.
+        single_value_upper_limit (bool):
+            Mandates that the highest bin should be single valued, with a small
+            precision tolerance, defined as 1.0E-6. The bin is thus (1 - 1.0E-6) to 1.
```
Is this "(1-1.0E-6) to 1" right?

bayliffe commented on improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py:
```diff
@@ -173,13 +181,15 @@ def test_without_single_value_limits():
-    def test_with_single_value_limits():
-        """Test the generation of probability bins with single value end
+    def test_with_both_single_value_limits():
+        """Test the generation of probability bins with both single value end
```
Done.

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/calibration/reliability_calibration.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5997286&bid=19663972)

bayliffe commented:
@Kat-90 thanks for the suggestion about another acceptance test. It made me notice that I hadn't actually given anyway to turn off the single value bins. This is because in clize, using the ```--single-value-lower-limit``` option for example sets this argument to true. I had also set the default value to True in the CLI, so basically it was always true. To keep everything nicely aligned, i.e. the behaviour of using the CLI with default options and the behaviour of importing and using the plugin with default options, I have modified both the CLI and plugin to have these arguments set to False by default.

I have subsequently tweaked the unit tests, added an extra one for the ```_create_probability_bins_coord``` method, and added the acceptance test you proposed.

@gavinevans you might want to take another look at this as well.
